views/js/runner/plugins/content/dialog/dialog.js

```diff
@@ -40,13 +41,64 @@ define([
          * Initialize the plugin (called during runner's init)
          */
         init : function init(){
-            var self = this;
             var testRunner = this.getTestRunner();
+            var alerts = [];
+            var confirms = [];
+
+            function addHandle(namespace, stack, dialog) {
+                stack.push({
+                    context: namespace,
+                    dialog: dialog
+                });
+
+                dialog.on('closed.modal', function() {
+                    removeHandle(stack, dialog);
+                });
+            }
+
+            function removeHandle(stack, dialog) {
+                _.forEach(stack, function(handle, index) {
+                    if (handle && dialog && dialog === handle.dialog) {
+                        stack[index] = null;
```

Comment:
Is there a reason why you set the entry to `null` and then doing a `cleanup` ? Do you need to keep the indexes ? Otherwise, you can use `_.remove`